edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ServiceIdentityTree.cs

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Core
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity.Service;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Concurrency;
+
+    /// <summary>
+    /// This is a tree implementation that uses the ParentScopes property
+    /// to reconstruct the hierarchy of modules/devices under the current
+    /// Edge device in a nested Edge environment.
+    /// </summary>
+    public class ServiceIdentityTree :
+        IServiceIdentityTree,
+        IAuthenticationChainProvider
```

Comment:
Do we need this another interface? IServiceIdentityTree seems to already implement GetAuthTree